### PR TITLE
Convert baseline model to pytorch. Add loss registry.

### PR DESCRIPTION
**Pull request by jeswan**
This PR adds the initial baseline model as well as a loss registry. The loss registry should be used to create custom losses as these will be immediately added to the baseline model. See comments in code for additional details.

Currently only one loss is implemented in the registry. Other losses can easily be added by adding them with the decorator (see comments in code).

### CODE REVIEW

File: src/losses/custom_loss.py [hunk 1]
```diff
@@ -18,7 +25,8 @@ def forward(self, y_true: torch.Tensor, y_pred: torch.Tensor) -> torch.Tensor:
         return self.l1(y_true, y_pred) + self.perceptual(y_true, y_pred)
 
 
-class L1_SSIM(nn.Module):
+@LossRegistry.register('l1_ssim')
```
Comment (jeswan) [inline src/losses/custom_loss.py:32]: Decorator to add loss to registry ('l1_ssim` is key. the value is the L1_SSIM class).

File: src/models/baseline/baseline_model.py [hunk 2]
```diff
@@ -101,29 +56,67 @@ def fit(self, dataloader: DataLoader):
...
-    # train
-    min_mae = model.fit(dm.val_dataloader())
-    global_disparity = model.global_disparity
-
+    model = BaselineModel().to(device)
+    model.eval()
+
+    loss_module_dict = {}
+    loss_result_dict = defaultdict(lambda: defaultdict(float))
+    losses_registry = LossRegistry.get_registry()
```
Comment (jeswan) [inline src/models/baseline/baseline_model.py:81]: Example to get the loss registry. The registry is a dict mapping the loss name (string) to the class of the loss. This allows us to easily add new losses without modifying the baseline code. This may be useful in other models as well.
